### PR TITLE
sched: start dynamicInformerFactory along with regular informerFactory

### PR DESCRIPTION
**Pull request by Huang-Wei**
#### What type of PR is this?

/kind cleanup
/sig scheduling

#### What this PR does / why we need it:

Followup of https://github.com/kubernetes/kubernetes/pull/104793#discussion_r706873499.

This PR moves the logic of starting dynamicInformerFactory right after regular informerFactory's start, and thus make the behavior intuitive and consistent.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Comment by Huang-Wei** [comment]
/hold

Built on top of #104793. Will rebase once #104793 gets merged.

File: test/integration/scheduler/queue_test.go
Comment (alculquicondor) [inline test/integration/scheduler/queue_test.go:339]: how is this related? Perhaps it's worth its own PR?

Comment (Huang-Wei) [reply test/integration/scheduler/queue_test.go:339]: > Perhaps it's worth its own PR?

SG. I will make it a separate one.

> how is this related?

Not quit. It's added to speed up surfacing the failure case of this integration test. For example, if a PR introduces a bug, the integration test won't pass for sure, but it hanged there (due to `Pop()` 'd hang), and time out when the global integration test's timeout (10m) is reached.

File: cmd/kube-scheduler/app/server.go
Comment (alculquicondor) [inline cmd/kube-scheduler/app/server.go:187]: We should do both `Start`s first and then `WaitForCacheSync`. Otherwise we are serializing the syncs.

Here and everywhere else.

Comment (Huang-Wei) [reply cmd/kube-scheduler/app/server.go:187]: I don't get it. It should suffice to ensure for each informerFactory, its Start() occurs before its WaitForCacheSync(). In other words, what's the side effect of starting dynInformerFactory after regular informerFactory's sync?

Comment (alculquicondor) [reply cmd/kube-scheduler/app/server.go:187]: My recommendation is not about order guarantees. What you have works.

However, it might be slightly faster if you first start all the workers and then wait for the sync of both. Rather than start the one informer, wait for the sync and then start the second informer.

Comment (Huang-Wei) [reply cmd/kube-scheduler/app/server.go]: > it might be slightly faster if you first start all the workers and then wait for the sync of both

That sounds fair.

BTW: I will append a commit, and recently I learned we can label a PR with command `/label tide/merge-method-squash`, so it will be auto-squashed on the fly and then merged into the git tree (but in the Github GUI, the PR still shows 2 commits). Let's give it a try.

**Review by alculquicondor — APPROVED** [review]

**Comment by Huang-Wei** [comment]
I will see if @neolit123 is willing to give on hand on reviewing the `test` changes.

/assign @neolit123

File: test/integration/util/util.go
Comment (neolit123) [inline test/integration/util/util.go:205]: this can be:

```suggestion
	testCtx.InformerFactory.Start(testCtx.Ctx.Done())
	testCtx.InformerFactory.WaitForCacheSync(testCtx.Ctx.Done())
	if testCtx.DynInformerFactory != nil {
		testCtx.DynInformerFactory.Start(testCtx.Ctx.Done())
		testCtx.DynInformerFactory.WaitForCacheSync(testCtx.Ctx.Done())
	}
```

**Review by neolit123 — COMMENTED** [review]
added one minor comment, but approve for /test.

/approve

File: test/integration/util/util.go
Comment (Huang-Wei) [reply test/integration/util/util.go:205]: This was intentional :) discussed in https://github.com/kubernetes/kubernetes/pull/105016#discussion_r709603422

Comment (neolit123) [reply test/integration/util/util.go:205]: I haven't played with that, so might be significant for bigger caches.

**Pull request merged**
